### PR TITLE
Add `onError` HOC

### LINKED ISSUE #9: Add `onError` HOC
Hooks into new (React 16) `componentDidCatch` lifecycle method. Should be backwards compatible with React 15, print warning if not supported.

### PR DESCRIPTION
**Pull request by rkillackey**
Resolves #9 

Also updates to React 16, along with any required dependency updates. 😅

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
   "devDependencies": {
     "babel-cli": "^6.22.2",
     "babel-eslint": "^7.1.1",
-    "babel-jest": "^18.0.0",
+    "babel-jest": "^21.2.0",
```
Comment (dpikt) [inline package.json]: I think this included in the latest jest so we can remove this dependency.

File: src/lifecycle/on-error.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *  }
+ *
+ *  export default onError(onComponentDidCatch)(MyComponent)
+ *
+**/
+
+export default function onError (onComponentDidCatch) {
+  
+  const SUPPORTED_REACT_VERSION = 16
+  const CURRENT_REACT_VERSION = React.version
```
Comment (dpikt) [inline src/lifecycle/on-error.js]: Instead of using parseInt, you can convert to a number here with `Number()`

File: src/lifecycle/on-error.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ *    logErrorToMyService(error, info)
+ *  }
+ *
+ *  export default onError(onComponentDidCatch)(MyComponent)
+ *
+**/
+
+export default function onError (onComponentDidCatch) {
+  
+  const SUPPORTED_REACT_VERSION = 16
```
Comment (dpikt) [inline src/lifecycle/on-error.js]: Can these declarations live outside of the function?

File: docs.md [hunk 4]
```diff
@@ -75,11 +76,42 @@ export default compose(
 
 Returns **[Function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function)** A HOC that can be used to wrap a component.
 
+## SUPPORTED_REACT_VERSION
```
Comment (dpikt) [inline docs.md]: 🤔 🤔 🤔 

Haha you'll probably have to use the @name property in the doc comment.

Comment (rkillackey) [reply docs.md]: whoa, weird!

**Pull request merged**